### PR TITLE
maxim_irq.c Multiple TMR interrupts and callbacks

### PR DESCRIPTION
**Pull request by danmois**
Add the possibility of using multiple interrupt callbacks for different timers.

Tested on MAX32655.

### CODE REVIEW

File: drivers/platform/maxim/max32650/maxim_irq.c [hunk 1]
```diff
@@ -388,16 +388,21 @@ int32_t max_irq_register_callback(struct no_os_irq_ctrl_desc *desc,
 		break;
 
 	case NO_OS_TIM_IRQ:
-		if (_events[NO_OS_EVT_RTC].actions == NULL) {
+		if (_events[NO_OS_EVT_TIM_ELAPSED].actions == NULL) {
```
Comment (CiprianRegus) [inline drivers/platform/maxim/max32650/maxim_irq.c:391]: I would split this part in a separate commit, since it's a fix. Otherwise, it looks good.

**Review by CiprianRegus — APPROVED** [review]

**Comment by danmois** [comment]
V2

- split commit to separate different fixes

**Review by amiclaus — APPROVED** [review]
